### PR TITLE
User's `qt.deploy.flags` should have priority over defaults

### PR DESCRIPTION
**Pull request by yh-sb**
#5350

### CODE REVIEW

File: xmake/rules/qt/install/mingw.lua [hunk 1]
```diff
@@ -81,18 +81,21 @@ function main(target, opt)
         table.insert(argv, "--verbose=0")
     end
 
-    if is_mode("debug") then
-        table.insert(argv, "--debug")
-    else
-        table.insert(argv, "--release")
+    -- make sure user flags have priority over default
+    local user_flags = target:values("qt.deploy.flags")
+    if not user_flags or not (user_flags:find("--release") or user_flags:find("--debug")) then
```
[the remark this answers is not in the document]
Comment (yh-sb) [reply xmake/rules/qt/install/mingw.lua]: Fixed.

File: xmake/rules/qt/install/windows.lua [hunk 2]
```diff
@@ -91,18 +91,21 @@ function main(target, opt)
         table.insert(argv, "--verbose=0")
     end
 
-    if is_mode("debug") then
-        table.insert(argv, "--debug")
-    else
-        table.insert(argv, "--release")
+    -- make sure user flags have priority over default
+    local user_flags = target:values("qt.deploy.flags")
+    if not user_flags or not (user_flags:find("--release") or user_flags:find("--debug")) then
```
[the remark this answers is not in the document]
Comment (yh-sb) [reply xmake/rules/qt/install/windows.lua]: Fixed.

**Comment by jingkaimori** [comment]
@waruqi @yh-sb value of `qt.deploy.flags` may be array, but there is no `table:find` method.

```
> xmake l
xmake> ({"-print"}):find("-p", 1, true) 
stdin:1: method 'find' is not callable (a nil value)
stack traceback:
        stdin:1: in main chunk
        [C]: in function 'sandbox/sandbox.interactive'
        ...dir\core\sandbox\modules\import\core\sandbox\sandbox.lua:102: in field 'interactive'
        @programdir\plugins\lua\main.lua:183: in function <@programdir\plugins\lua\main.lua:161>
        (...tail calls...)
        [C]: in function 'xpcall'
        @programdir\core\base\utils.lua:275: in function 'base/utils.trycall'
        (...tail calls...)
        @programdir\core\base\task.lua:491: in function 'base/task.run'
        @programdir\core\main.lua:327: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
```

thus the follow code will throw error when running `xmake install`:
```lua
add_rules("mode.debug", "mode.release")

add_requires("qt5widgets")

target("testqt2")
    add_rules("qt.widgetapp")
    set_values("qt.deploy.flags", {"-printsupport", "--no-opengl-sw", "--no-translations"})
    add_headerfiles("src/*.h")
    add_files("src/*.cpp")
    add_files("src/mainwindow.ui")
    -- add files with Q_OBJECT meta (only for qt.moc)
    add_files("src/mainwindow.h")
    add_packages("qt5widgets")
```

**Comment by waruqi** [comment]
> @waruqi @yh-sb value of `qt.deploy.flags` may be array, but there is no `table:find` method.
> 
> ```
> > xmake l
> xmake> ({"-print"}):find("-p", 1, true) 
> stdin:1: method 'find' is not callable (a nil value)
> stack traceback:
>         stdin:1: in main chunk
>         [C]: in function 'sandbox/sandbox.interactive'
>         ...dir\core\sandbox\modules\import\core\sandbox\sandbox.lua:102: in field 'interactive'
>         @programdir\plugins\lua\main.lua:183: in function <@programdir\plugins\lua\main.lua:161>
>         (...tail calls...)
>         [C]: in function 'xpcall'
>         @programdir\core\base\utils.lua:275: in function 'base/utils.trycall'
>         (...tail calls...)
>         @programdir\core\base\task.lua:491: in function 'base/task.run'
>         @programdir\core\main.lua:327: in upvalue 'cotask'
>         @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
> ```
> 
> thus the follow code will throw error when running `xmake install`:
> 
> ```lua
> add_rules("mode.debug", "mode.release")
> 
> add_requires("qt5widgets")
> 
> target("testqt2")
>     add_rules("qt.widgetapp")
>     set_values("qt.deploy.flags", {"-printsupport", "--no-opengl-sw", "--no-translations"})
>     add_headerfiles("src/*.h")
>     add_files("src/*.cpp")
>     add_files("src/mainwindow.ui")
>     -- add files with Q_OBJECT meta (only for qt.moc)
>     add_files("src/mainwindow.h")
>     add_packages("qt5widgets")
> ```

you can open a pr to improve it.

**Pull request merged**